diff -Naur a/Makefile b/Makefile
--- a/Makefile	2014-07-24 21:02:19.000000000 +0300
+++ b/Makefile	2014-07-24 21:54:00.552233231 +0300
@@ -21,24 +21,13 @@
 ### The directory environment:
 
 # Use package data if installed...otherwise assume we're under the VDR source directory:
-PKGCFG = $(if $(VDRDIR),$(shell pkg-config --variable=$(1) $(VDRDIR)/vdr.pc),$(shell pkg-config --variable=$(1) vdr || pkg-config --variable=$(1) ../../../vdr.pc))
-LIBDIR = $(call PKGCFG,libdir)
-LOCDIR = $(call PKGCFG,locdir)
-PLGCFG  = $(call PKGCFG,plgcfg)
-VDRCONFDIR = $(call PKGCFG,configdir)
-PLGRESDIR = $(call PKGCFG,resdir)/plugins/$(PLUGIN)
 TMPDIR ?= /tmp
 
 ### The compiler options:
-export CFLAGS   = $(call PKGCFG,cflags)
-export CXXFLAGS = $(call PKGCFG,cxxflags)
-
-### Allow user defined options to overwrite defaults:
-
--include $(PLGCFG)
+include $(VDRDIR)/Make.global
 
 ### The version number of VDR's plugin API:
-APIVERSION = $(call PKGCFG,apiversion)
+APIVERSION = $(shell sed -ne '/define APIVERSION/s/^.*"\(.*\)".*$$/\1/p' $(VDRDIR)/config.h)
 
 ### The name of the distribution archive:
 
@@ -51,6 +40,7 @@
 
 ### Includes and Defines and Dependencies (add further entries here):
 
+INCLUDES += -I$(VDRDIR)/include
 DEFINES += -DPLUGIN_NAME_I18N='"$(PLUGIN)"'
 
 ifeq ($(IMAGELIB), imagemagick)
@@ -114,6 +104,7 @@
 
 $(SOFILE): $(OBJS)
 	$(CXX) $(CXXFLAGS) $(LDFLAGS) -shared $(OBJS) $(LIBS) -o $@
+	@cp --remove-destination $@ $(LIBDIR)/$@.$(APIVERSION)
 
 install-lib: $(SOFILE)
 	install -D $^ $(DESTDIR)$(LIBDIR)/$^.$(APIVERSION)
