diff -Naur a/Makefile b/Makefile
--- a/Makefile	2018-02-27 20:05:03.000000000 +0200
+++ b/Makefile	2018-04-17 20:55:12.238889530 +0300
@@ -14,21 +14,15 @@
 ### The directory environment:
 
 # Use package data if installed...otherwise assume we're under the VDR source directory:
-PKGCFG = $(if $(VDRDIR),$(shell pkg-config --variable=$(1) $(VDRDIR)/vdr.pc),$(shell pkg-config --variable=$(1) vdr || pkg-config --variable=$(1) ../../../vdr.pc))
-LIBDIR = $(call PKGCFG,libdir)
-LOCDIR = $(call PKGCFG,locdir)
-PLGCFG = $(call PKGCFG,plgcfg)
-#
 TMPDIR ?= /tmp
 
 ### The compiler options:
 
-export CFLAGS   = $(call PKGCFG,cflags)
-export CXXFLAGS = $(call PKGCFG,cxxflags)
+include $(VDRDIR)/Make.global
 
 ### The version number of VDR's plugin API:
 
-APIVERSION = $(call PKGCFG,apiversion)
+APIVERSION = $(shell sed -ne '/define APIVERSION/s/^.*"\(.*\)".*$$/\1/p' $(VDRDIR)/config.h)
 
 ### Allow user defined options to overwrite defaults:
 
@@ -45,11 +39,17 @@
 
 ### Includes and Defines (add further entries here):
 
+INCLUDES += -I$(VDRDIR)/include
+
 INCLUDES += `pkg-config libxml++-2.6 --cflags`
 INCLUDES += `pkg-config glibmm-2.4 --cflags`
+INCLUDES += `pkg-config glib-2.0 --cflags`
 
 LIBS +=  `pkg-config libxml++-2.6 --libs`
 LIBS +=  `pkg-config glibmm-2.4 --libs`
+LIBS +=  `pkg-config glib-2.0 --libs`
+
+CXXFLAGS +=-std=c++11
 
 DEFINES += -DPLUGIN_NAME_I18N='"$(PLUGIN)"' \
 	$(if $(GIT_REV), -DGIT_REV='"$(GIT_REV)"')
@@ -110,6 +110,7 @@
 
 $(SOFILE): $(OBJS)
 	$(CXX) $(CPPFLAGS) $(CXXFLAGS) $(LDFLAGS) -shared $(OBJS) $(LIBS) -o $@
+	@cp --remove-destination $@ $(LIBDIR)/$@.$(APIVERSION)
 
 install-lib: $(SOFILE)
 	install -D $^ $(DESTDIR)$(LIBDIR)/$^.$(APIVERSION)
