diff -Naur a/linux/drivers/media/video/ivtv/ivtv-driver.c b/linux/drivers/media/video/ivtv/ivtv-driver.c
--- a/linux/drivers/media/video/ivtv/ivtv-driver.c	2016-11-09 23:33:29.000000000 +0200
+++ b/linux/drivers/media/video/ivtv/ivtv-driver.c	2016-12-25 18:52:20.031422682 +0200
@@ -738,6 +738,7 @@
 
 	INIT_WORK(&itv->irq_work_queue, ivtv_irq_work_handler);
 #else
+#if LINUX_VERSION_CODE < KERNEL_VERSION(4, 9, 0)
 	init_kthread_worker(&itv->irq_worker);
 	itv->irq_worker_task = kthread_run(kthread_worker_fn, &itv->irq_worker,
 					   itv->v4l2_dev.name);
@@ -749,6 +750,19 @@
 	sched_setscheduler(itv->irq_worker_task, SCHED_FIFO, &param);
 
 	init_kthread_work(&itv->irq_work, ivtv_irq_work_handler);
+#else
+	kthread_init_worker(&itv->irq_worker);
+	itv->irq_worker_task = kthread_run(kthread_worker_fn, &itv->irq_worker,
+					   itv->v4l2_dev.name);
+	if (IS_ERR(itv->irq_worker_task)) {
+		IVTV_ERR("Could not create ivtv task\n");
+		return -1;
+	}
+	/* must use the FIFO scheduler as it is realtime sensitive */
+	sched_setscheduler(itv->irq_worker_task, SCHED_FIFO, &param);
+
+	kthread_init_work(&itv->irq_work, ivtv_irq_work_handler);
+#endif
 #endif
 
 	/* start counting open_id at 1 */
@@ -1428,9 +1442,15 @@
 	flush_workqueue(itv->irq_work_queues);
 	destroy_workqueue(itv->irq_work_queues);
 #else
+#if LINUX_VERSION_CODE < KERNEL_VERSION(4, 9, 0)
 	/* Kill irq worker */
 	flush_kthread_worker(&itv->irq_worker);
 	kthread_stop(itv->irq_worker_task);
+#else
+	/* Kill irq worker */
+	kthread_flush_worker(&itv->irq_worker);
+	kthread_stop(itv->irq_worker_task);
+#endif
 #endif
 
 	ivtv_streams_cleanup(itv, 1);
diff -Naur a/linux/drivers/media/video/ivtv/ivtv-irq.c b/linux/drivers/media/video/ivtv/ivtv-irq.c
--- a/linux/drivers/media/video/ivtv/ivtv-irq.c	2016-11-09 23:33:29.000000000 +0200
+++ b/linux/drivers/media/video/ivtv/ivtv-irq.c	2016-12-25 18:58:57.840221341 +0200
@@ -1033,7 +1033,11 @@
 #if LINUX_VERSION_CODE <= KERNEL_VERSION(2,6,35)
 		queue_work(itv->irq_work_queues, &itv->irq_work_queue);
 #else
+#if LINUX_VERSION_CODE < KERNEL_VERSION(4,9,0)
 		queue_kthread_work(&itv->irq_worker, &itv->irq_work);
+#else
+		kthread_queue_work(&itv->irq_worker, &itv->irq_work);
+#endif
 #endif
 	}
 
diff -Naur a/linux/drivers/media/video/ivtv/ivtv-udma.c b/linux/drivers/media/video/ivtv/ivtv-udma.c
--- a/linux/drivers/media/video/ivtv/ivtv-udma.c	2016-11-09 23:33:29.000000000 +0200
+++ b/linux/drivers/media/video/ivtv/ivtv-udma.c	2016-12-25 19:10:28.834666134 +0200
@@ -133,12 +133,15 @@
 
 	/* Get user pages for DMA Xfer */
 	down_read(&current->mm->mmap_sem);
-	#if LINUX_VERSION_CODE < KERNEL_VERSION(4, 6, 0)
-	err = get_user_pages(current, current->mm,
-			user_dma.uaddr, user_dma.page_count, 0, 1, dma->map, NULL);
-	#else
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(4, 9, 0)
+	err = get_user_pages(user_dma.uaddr, user_dma.page_count, 0, dma->map, NULL);
+#else
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(4, 6, 0)
 	err = get_user_pages(user_dma.uaddr, user_dma.page_count, 0, 1, dma->map, NULL);
-	#endif	
+#else
+	err = get_user_pages(current, current->mm, user_dma.uaddr, user_dma.page_count, 0, 1, dma->map, NULL);
+#endif
+#endif
 	
 	up_read(&current->mm->mmap_sem);
 
diff -Naur a/linux/drivers/media/video/ivtv/ivtv-yuv.c b/linux/drivers/media/video/ivtv/ivtv-yuv.c
--- a/linux/drivers/media/video/ivtv/ivtv-yuv.c	2016-11-09 23:33:29.000000000 +0200
+++ b/linux/drivers/media/video/ivtv/ivtv-yuv.c	2016-12-25 19:20:48.963748946 +0200
@@ -76,24 +76,28 @@
 
 	/* Get user pages for DMA Xfer */
 	down_read(&current->mm->mmap_sem);
-	#if LINUX_VERSION_CODE < KERNEL_VERSION(4, 6, 0)
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(4, 9, 0)
+	y_pages = get_user_pages(y_dma.uaddr, y_dma.page_count, 0, &dma->map[0], NULL);
+#else
+#if LINUX_VERSION_CODE < KERNEL_VERSION(4, 6, 0)
 	y_pages = get_user_pages(current, current->mm, y_dma.uaddr, y_dma.page_count, 0, 1, &dma->map[0], NULL);
-	#else
+#else
 	y_pages = get_user_pages(y_dma.uaddr, y_dma.page_count, 0, 1, &dma->map[0], NULL);
-	#endif	
+#endif
+#endif
 	
 	uv_pages = 0; /* silence gcc. value is set and consumed only if: */
 	if (y_pages == y_dma.page_count) {
 
-		#if LINUX_VERSION_CODE < KERNEL_VERSION(4, 6, 0)
-			uv_pages = get_user_pages(current, current->mm,
-					  uv_dma.uaddr, uv_dma.page_count, 0, 1,
-					  &dma->map[y_pages], NULL);
-
-		#else
-			uv_pages = get_user_pages(uv_dma.uaddr, uv_dma.page_count, 0, 1,
-					  &dma->map[y_pages], NULL);
-		#endif	
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(4, 9, 0)
+		uv_pages = get_user_pages(uv_dma.uaddr, uv_dma.page_count, 0, &dma->map[y_pages], NULL);
+#else
+#if LINUX_VERSION_CODE < KERNEL_VERSION(4, 6, 0)
+		uv_pages = get_user_pages(current, current->mm, uv_dma.uaddr, uv_dma.page_count, 0, 1, &dma->map[y_pages], NULL);
+#else
+		uv_pages = get_user_pages(uv_dma.uaddr, uv_dma.page_count, 0, 1, &dma->map[y_pages], NULL);
+#endif
+#endif
 
 	}
 	up_read(&current->mm->mmap_sem);
diff -Naur a/linux/drivers/media/video/videobuf2-dma-sg.c b/linux/drivers/media/video/videobuf2-dma-sg.c
--- a/linux/drivers/media/video/videobuf2-dma-sg.c	2016-11-09 23:33:29.000000000 +0200
+++ b/linux/drivers/media/video/videobuf2-dma-sg.c	2016-12-25 17:53:23.101359886 +0200
@@ -148,24 +148,15 @@
 
 	down_read(&current->mm->mmap_sem);
 
-#if LINUX_VERSION_CODE < KERNEL_VERSION(4, 6, 0)
-			num_pages_from_user = get_user_pages(current, current->mm,
-								 vaddr & PAGE_MASK,
-								 buf->sg_desc.num_pages,
-								 write,
-								 1, /* force */
-								 buf->pages,
-								 NULL);
-
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(4, 9, 0)
+	num_pages_from_user = get_user_pages(vaddr & PAGE_MASK, buf->sg_desc.num_pages, write, buf->pages, NULL);
 #else
-			num_pages_from_user = get_user_pages(vaddr & PAGE_MASK,
-								 buf->sg_desc.num_pages,
-								 write,
-								 1, /* force */
-								 buf->pages,NULL);
-#endif	
-
-	
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(4, 6, 0)
+	num_pages_from_user = get_user_pages(vaddr & PAGE_MASK, buf->sg_desc.num_pages, write, 1, buf->pages, NULL);
+#else
+	num_pages_from_user = get_user_pages(current, current->mm, vaddr & PAGE_MASK, buf->sg_desc.num_pages, write, 1, buf->pages, NULL);
+#endif
+#endif
 
 	up_read(&current->mm->mmap_sem);
 	if (num_pages_from_user != buf->sg_desc.num_pages)
diff -Naur a/linux/drivers/media/video/videobuf-dma-sg.c b/linux/drivers/media/video/videobuf-dma-sg.c
--- a/linux/drivers/media/video/videobuf-dma-sg.c	2016-11-09 23:33:29.000000000 +0200
+++ b/linux/drivers/media/video/videobuf-dma-sg.c	2016-12-25 17:53:56.645884249 +0200
@@ -182,17 +182,15 @@
 	dprintk(1, "init user [0x%lx+0x%lx => %d pages]\n",
 		data, size, dma->nr_pages);
 
-	#if LINUX_VERSION_CODE < KERNEL_VERSION(4, 6, 0)
-			err = get_user_pages(current, current->mm,
-			     data & PAGE_MASK, dma->nr_pages,
-			     rw == READ, 1, /* force */
-			     dma->pages, NULL);
-	#else
-			err = get_user_pages(data & PAGE_MASK, dma->nr_pages,
-			     rw == READ, 1, /* force */
-			     dma->pages,NULL);
-	#endif
-
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(4, 9, 0)
+	err = get_user_pages(data & PAGE_MASK, dma->nr_pages, rw == READ, dma->pages, NULL);
+#else
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(4, 6, 0)
+	err = get_user_pages(data & PAGE_MASK, dma->nr_pages, rw == READ, 1, dma->pages, NULL);
+#else
+	err = get_user_pages(current, current->mm,data & PAGE_MASK, dma->nr_pages, rw == READ, 1, dma->pages, NULL);
+#endif
+#endif
 
 	if (err != dma->nr_pages) {
 		dma->nr_pages = (err >= 0) ? err : 0;
