diff -Naur a/CMakeLists.txt b/CMakeLists.txt
--- a/CMakeLists.txt	2019-01-20 08:56:22.937150542 +0200
+++ b/CMakeLists.txt	2019-01-20 09:02:38.175849955 +0200
@@ -498,7 +498,7 @@
 
 set (exe_name "oscam")
 add_executable (${exe_name} ${exe_srcs} ${exe_hdrs})
-target_link_libraries (${exe_name} ${csoscam} ${csmodules} ${csreaders} csctapi cscrypt minilzo)
+target_link_libraries (${exe_name} ${csoscam} ${csmodules} ${csreaders} csctapi cscrypt minilzo udev)
 if(HAVE_LIBRT AND HAVE_LIBUSB)
 	if (LIBUSBDIR)
 		set (libusb_link "imp_libusb")
diff -Naur a/utils/CMakeLists.txt b/utils/CMakeLists.txt
--- a/utils/CMakeLists.txt	2019-01-20 08:56:22.853149932 +0200
+++ b/utils/CMakeLists.txt	2019-01-20 09:03:14.768111216 +0200
@@ -12,7 +12,7 @@
 
 set (util_name "list_smargo")
 add_executable (${util_name} ${exe_srcs} ${exe_hdrs})
-target_link_libraries (${util_name} ${libusb_link} ${rt_link} ${setupapi_link} ${ole32_link} ${shell32_link} ${pthread_link} ${dl_link})
+target_link_libraries (${util_name} ${libusb_link} ${rt_link} ${setupapi_link} ${ole32_link} ${shell32_link} ${pthread_link} ${dl_link} udev)
 #----------------------- printout resume -----------------------------
 
 message (STATUS "Utils: operating system: ${OSCamOperatingSystem}")
