diff -Naur a/CMakeLists.txt b/CMakeLists.txt
--- a/CMakeLists.txt	2019-01-20 08:56:22.937150542 +0200
+++ b/CMakeLists.txt	2019-01-20 09:46:37.045810926 +0200
@@ -561,6 +561,7 @@
 if (NOT OSCamOperatingSystem MATCHES "Mac OS X")
 if (NOT OSCamOperatingSystem MATCHES "Windows/Cygwin")
 	target_link_libraries (${exe_name} pcsclite)
+	target_link_libraries (${exe_name} pthread)
 endif (NOT OSCamOperatingSystem MATCHES "Windows/Cygwin")
 endif (NOT OSCamOperatingSystem MATCHES "Mac OS X")
 endif (HAVE_PCSC)
