diff -Naur a/qtbase/mkspecs/common/clang.conf b/qtbase/mkspecs/common/clang.conf
--- a/qtbase/mkspecs/common/clang.conf	2019-01-13 07:12:28.688146226 +0200
+++ b/qtbase/mkspecs/common/clang.conf	2019-01-13 07:44:58.965575245 +0200
@@ -16,7 +16,6 @@
 CONFIG                 += clang_pch_style
 QMAKE_PCH_OUTPUT_EXT    = .pch
 
-QMAKE_CFLAGS_ISYSTEM             = -isystem
 QMAKE_CFLAGS_PRECOMPILE          = -x c-header -c ${QMAKE_PCH_INPUT} -o ${QMAKE_PCH_OUTPUT}
 QMAKE_CFLAGS_USE_PRECOMPILE      = -Xclang -include-pch -Xclang ${QMAKE_PCH_OUTPUT}
 QMAKE_CFLAGS_LTCG                = -flto
diff -Naur a/qtbase/mkspecs/common/gcc-base.conf b/qtbase/mkspecs/common/gcc-base.conf
--- a/qtbase/mkspecs/common/gcc-base.conf	2019-01-13 07:12:28.692146278 +0200
+++ b/qtbase/mkspecs/common/gcc-base.conf	2019-01-13 07:44:58.965575245 +0200
@@ -46,7 +46,6 @@
 QMAKE_CFLAGS_SHLIB         += $$QMAKE_CFLAGS_PIC
 QMAKE_CFLAGS_STATIC_LIB    += $$QMAKE_CFLAGS_PIC
 QMAKE_CFLAGS_APP           += $$QMAKE_CFLAGS_PIC
-QMAKE_CFLAGS_ISYSTEM        = -isystem
 QMAKE_CFLAGS_YACC          += -Wno-unused -Wno-parentheses
 QMAKE_CFLAGS_HIDESYMS      += -fvisibility=hidden
 QMAKE_CFLAGS_EXCEPTIONS_OFF += -fno-exceptions
diff -Naur a/qtbase/mkspecs/linux-icc/qmake.conf b/qtbase/mkspecs/linux-icc/qmake.conf
--- a/qtbase/mkspecs/linux-icc/qmake.conf	2019-01-13 07:12:28.784147468 +0200
+++ b/qtbase/mkspecs/linux-icc/qmake.conf	2019-01-13 07:44:58.965575245 +0200
@@ -9,7 +9,6 @@
 # modifications to icc-base-unix.conf
 
 QMAKE_CFLAGS_YACC       =
-QMAKE_CFLAGS_ISYSTEM    = -isystem
 QMAKE_CFLAGS_THREAD     = -D_REENTRANT
 
 QMAKE_CXXFLAGS_YACC     = $$QMAKE_CFLAGS_YACC
diff -Naur a/qtbase/qmake/generators/unix/unixmake2.cpp b/qtbase/qmake/generators/unix/unixmake2.cpp
--- a/qtbase/qmake/generators/unix/unixmake2.cpp	2019-01-13 07:12:23.856083597 +0200
+++ b/qtbase/qmake/generators/unix/unixmake2.cpp	2019-01-13 07:44:58.969575247 +0200
@@ -1413,10 +1413,8 @@
     }
 
     if (include_deps && project->isActiveConfig("gcc_MD_depends")) {
-        // use -MMD if we know about -isystem too
-        ProString MD_flag(project->values("QMAKE_CFLAGS_ISYSTEM").isEmpty() ? "-MD" : "-MMD");
-        project->values("QMAKE_CFLAGS") += MD_flag;
-        project->values("QMAKE_CXXFLAGS") += MD_flag;
+        project->values("QMAKE_CFLAGS") += "-MD";
+        project->values("QMAKE_CXXFLAGS") += "-MD";
     }
 }
 
diff -Naur a/qtbase/qmake/generators/win32/mingw_make.cpp b/qtbase/qmake/generators/win32/mingw_make.cpp
--- a/qtbase/qmake/generators/win32/mingw_make.cpp	2019-01-13 07:12:23.852083545 +0200
+++ b/qtbase/qmake/generators/win32/mingw_make.cpp	2019-01-13 07:44:58.969575247 +0200
@@ -262,17 +262,11 @@
 {
     t << "INCPATH       = ";
 
-    QString isystem = var("QMAKE_CFLAGS_ISYSTEM");
     const ProStringList &incs = project->values("INCLUDEPATH");
     for (ProStringList::ConstIterator incit = incs.begin(); incit != incs.end(); ++incit) {
         QString inc = (*incit).toQString();
         inc.replace(QRegExp("\\\\$"), "");
-
-        if (!isystem.isEmpty() && isSystemInclude(inc))
-            t << isystem << ' ';
-        else
-            t << "-I";
-        t << escapeFilePath(inc) << ' ';
+        t << "-I" << escapeFilePath(inc) << ' ';
     }
     t << endl;
 }