diff -Naur a/stdext.h b/stdext.h
--- a/stdext.h	2017-06-19 21:09:05.000000000 +0300
+++ b/stdext.h	2017-06-24 17:14:43.049076784 +0300
@@ -3,56 +3,38 @@
 
 #if __GNUC__ >= 4
 
-#	include <tr1/functional>
-#	include <tr1/memory>
+#if __GNUC__ >= 6
 
-#else
-
-#	include <boost/version.hpp>
-
-#	define BOOST_MAJOR_VERSION (BOOST_VERSION / 100000)
-#	define BOOST_MINOR_VERSION ((BOOST_VERSION / 100) % 1000)
-
-#	if BOOST_MAJOR_VERSION >= 1 && BOOST_MINOR_VERSION >= 34
-
-#		include <boost/tr1/functional.hpp>
-#		include <boost/tr1/memory.hpp>
-
-#	elif BOOST_MAJOR_VERSION >= 1 && BOOST_MINOR_VERSION >= 32
+#include <memory>
+#include <functional>
 
-#		include <boost/bind.hpp>
-#		include <boost/shared_ptr.hpp>
-#		include <boost/weak_ptr.hpp>
-
-namespace std { 
+namespace std {
 namespace tr1 {
 
-	using boost::bind;
-	using boost::shared_ptr;
-	using boost::weak_ptr;
-
-	namespace placeholders {
-		using ::_1;
-		using ::_2;
-		using ::_3;
-		using ::_4;
-		using ::_5;
-		using ::_6;
-		using ::_7;
-		using ::_8;
-		using ::_9;
-	}
-
-} // namespace std
+       using std::bind;
+       using std::shared_ptr;
+       using std::weak_ptr;
+
+       namespace placeholders {
+               using std::placeholders::_1;
+               using std::placeholders::_2;
+               using std::placeholders::_3;
+               using std::placeholders::_4;
+               using std::placeholders::_5;
+               using std::placeholders::_6;
+               using std::placeholders::_7;
+               using std::placeholders::_8;
+               using std::placeholders::_9;
+       }
 } // namespace tr1
+} // namespace std
 
-#	else
+#else
 
-#		error "Your Compiler is too old and you don't have boost >= 1.32.0 installed."
-#		error "Please either install boost 1.32.0 or higher (1.34.0 is recommended)"
-#		error "or upgrade your compiler suite to at least GCC 4.0"
+#include <tr1/functional>
+#include <tr1/memory>
 
-#	endif
+#endif
 
 #endif
 
