diff -Naur a/cmake/build_configurations/mysql_release.cmake b/cmake/build_configurations/mysql_release.cmake
--- a/cmake/build_configurations/mysql_release.cmake	2020-08-18 18:50:58.838153709 +0000
+++ b/cmake/build_configurations/mysql_release.cmake	2020-08-18 18:54:00.401370864 +0000
@@ -129,7 +129,7 @@
 
 IF(UNIX)
   SET(WITH_EXTRA_CHARSETS all CACHE STRING "")
-  SET(PLUGIN_AUTH_PAM YES CACHE BOOL "")
+  SET(PLUGIN_AUTH_PAM NO CACHE BOOL "")
 
   IF(CMAKE_SYSTEM_NAME STREQUAL "Linux")
     IF(NOT IGNORE_AIO_CHECK)
