diff -Naur a/osd.c b/osd.c
--- a/osd.c	2018-04-06 11:43:15.000000000 +0300
+++ b/osd.c	2018-04-17 20:38:53.695543571 +0300
@@ -2029,8 +2029,6 @@
         }
      return Osd;
      }
-  else
-     esyslog("ERROR: no OSD provider available - using dummy OSD!");
   return new cOsd(Left, Top, 999); // create a dummy cOsd, so that access won't result in a segfault
 }
 
diff -Naur a/vdr.c b/vdr.c
--- a/vdr.c	2018-04-10 16:24:43.000000000 +0300
+++ b/vdr.c	2018-04-17 20:39:48.395904542 +0300
@@ -768,12 +768,6 @@
   Folders.Load(AddDirectory(ConfigDirectory, "folders.conf"));
   CamResponsesLoad(AddDirectory(ConfigDirectory, "camresponses.conf"), true);
 
-  if (!*cFont::GetFontFileName(Setup.FontOsd)) {
-     const char *msg = "no fonts available - OSD will not show any text!";
-     fprintf(stderr, "vdr: %s\n", msg);
-     esyslog("ERROR: %s", msg);
-     }
-
   // Recordings:
 
   cRecordings::Update();
