diff -Naur a/configure b/configure
--- a/configure	2018-07-26 11:57:06.000000000 +0300
+++ b/configure	2018-08-06 15:54:59.209375057 +0300
@@ -159,20 +159,6 @@
   COMPILER=gcc
 fi
 
-check_cc_snippet strlcat '#include <string.h>
-int test(int argc, char **argv) {
-  char dst[10];
-  strlcat("test", dst, sizeof(dst));
-  return 0;
-}'
-
-check_cc_snippet strlcpy '#include <string.h>
-int test(int argc, char **argv) {
-  char dst[10];
-  strlcpy("test", dst, sizeof(dst));
-  return 0;
-}'
-
 check_cc_snippet getloadavg '#include <stdlib.h>
 void test() { getloadavg(NULL,0); }'
 
diff -Naur a/src/tvh_string.h b/src/tvh_string.h
--- a/src/tvh_string.h	2018-07-26 11:57:06.000000000 +0300
+++ b/src/tvh_string.h	2018-08-06 15:56:07.677375057 +0300
@@ -47,7 +47,6 @@
   return s1;
 }
 
-#ifndef ENABLE_STRLCPY
 static inline size_t strlcpy(char *dst, const char *src, size_t size)
 {
   size_t ret = strlen(src);
@@ -58,9 +57,7 @@
   }
   return ret;
 }
-#endif
 
-#ifndef ENABLE_STRLCAT
 static inline size_t strlcat(char *dst, const char *src, size_t count)
 {
   size_t dlen = strlen(dst);
@@ -75,7 +72,6 @@
   dst[len] = '\0';
   return res;
 }
-#endif
 
 #define tvh_strlcatf(buf, size, ptr, fmt...) \
   do { int __r = snprintf((buf) + ptr, (size) - ptr, fmt); \
