diff -Naur a/src/input/mpegts/scanfile.c b/src/input/mpegts/scanfile.c
--- a/src/input/mpegts/scanfile.c	2021-06-06 16:33:40.686983259 +0300
+++ b/src/input/mpegts/scanfile.c	2021-06-13 19:28:05.764659444 +0300
@@ -921,7 +921,7 @@
 #elif defined(PLATFORM_FREEBSD)
     path = "/usr/local/share/dtv-scan-tables";
 #else
-    path = "/usr/share/dvb";
+    path = "/usr/share/tvheadend/dvb-scan";
 #endif
 
   if (!initialized) {
