diff -Naur a/configure b/configure
--- a/configure	2021-06-13 18:16:26.678278454 +0300
+++ b/configure	2021-06-13 19:23:31.322569610 +0300
@@ -419,7 +419,7 @@
 else
 
   if enabled_or_auto hdhomerun_client; then
-    if check_cc_header 'libhdhomerun/hdhomerun' libhdhomerun; then
+    if check_cc_header 'hdhomerun' libhdhomerun; then
       enable  hdhomerun_client
       LDFLAGS="$LDFLAGS -lhdhomerun"
     fi
diff -Naur a/src/input/mpegts/tvhdhomerun/tvhdhomerun.c b/src/input/mpegts/tvhdhomerun/tvhdhomerun.c
--- a/src/input/mpegts/tvhdhomerun/tvhdhomerun.c	2021-06-06 16:33:40.742983764 +0300
+++ b/src/input/mpegts/tvhdhomerun/tvhdhomerun.c	2021-06-13 19:24:59.631243024 +0300
@@ -17,7 +17,7 @@
  *  along with this program.  If not, see <http://www.gnu.org/licenses/>.
  */
 
-#include "libhdhomerun/hdhomerun.h"
+#include "hdhomerun.h"
 
 #include "tvheadend.h"
 #include "input.h"
diff -Naur a/src/input/mpegts/tvhdhomerun/tvhdhomerun_frontend.c b/src/input/mpegts/tvhdhomerun/tvhdhomerun_frontend.c
--- a/src/input/mpegts/tvhdhomerun/tvhdhomerun_frontend.c	2021-06-06 16:33:40.742983764 +0300
+++ b/src/input/mpegts/tvhdhomerun/tvhdhomerun_frontend.c	2021-06-13 19:25:36.951527315 +0300
@@ -17,7 +17,7 @@
  *  along with this program.  If not, see <http://www.gnu.org/licenses/>.
  */
 
-#include "libhdhomerun/hdhomerun.h"
+#include "hdhomerun.h"
 
 #include <fcntl.h>
 #include "tvheadend.h"
