diff -Naur a/imageloader.c b/imageloader.c
--- a/imageloader.c	2014-10-18 00:39:10.000000000 +0300
+++ b/imageloader.c	2014-10-20 04:16:58.638680858 +0300
@@ -114,12 +114,27 @@
 bool cImageLoader::LoadImage(cString FileName, cString Path, cString Extension) {
     try {
         cString File = cString::sprintf("%s%s.%s", *Path, *FileName, *Extension);
-        dsyslog("imageloader: trying to load: %s", *File);
-        //printf("imageloader: trying to load: %s\n", *File);
+        #ifdef DEBUGIMAGELOADTIME
+            dsyslog("imageloader: trying to load: %s", *File);
+        #endif
         buffer.read(*File);
-        dsyslog("imageloader: %s sucessfully loaded", *File);
-        //printf("imageloader: %s sucessfully loaded\n", *File);
-    } catch (...) {     
+        #ifdef DEBUGIMAGELOADTIME
+            dsyslog("imageloader: %s sucessfully loaded", *File);
+        #endif
+    } catch( Magick::Warning &warning ) {
+        #ifdef DEBUGIMAGELOADTIME
+            dsyslog("imageloader: Magick Warning: %s", warning.what());
+        #endif
+        return true;
+    } catch( Magick::Error &error ) {
+        #ifdef DEBUGIMAGELOADTIME
+            dsyslog("imageloader: Magick Error: %s", error.what());
+        #endif
+        return false;
+    } catch(...) {
+        #ifdef DEBUGIMAGELOADTIME
+            dsyslog("imageloader: an unknown Magick error occured during image loading");
+        #endif
         return false;
     }
     return true;
