diff -Naur a/statemachine.c b/statemachine.c
--- a/statemachine.c	2018-11-24 19:33:13.834610089 +0200
+++ b/statemachine.c	2018-11-24 19:38:05.326602094 +0200
@@ -309,7 +309,9 @@
                  else
                     newState = eAddChannels;
               if (time(0) != tm) {
-                 MenuScanning->SetProgress(lProgress);
+                 MenuScanning->SetProgress(lProgress);
+                 if (MenuScanning)
+                    MenuScanning->SetProgress(lProgress);
                  tm = time(0);
                  }
               }
