diff -Naur a/v4l/Makefile b/v4l/Makefile
--- a/v4l/Makefile	2017-08-15 11:08:46.976363161 +0300
+++ b/v4l/Makefile	2017-08-15 11:14:09.122219715 +0300
@@ -51,7 +51,7 @@
 	@echo Kernel build directory is $(OUTDIR)
 	$(MAKE) -C ../linux apply_patches
 	$(MAKE) -C $(OUTDIR) SUBDIRS=$(PWD) $(MYCFLAGS) modules
-	./scripts/rmmod.pl check
+#	./scripts/rmmod.pl check
 #	$(MAKE) checkpatch
 
 mismatch:: prepare firmware
