diff -Naur a/Makefile.ffmpeg b/Makefile.ffmpeg
--- a/Makefile.ffmpeg	2017-07-06 16:14:55.000000000 +0300
+++ b/Makefile.ffmpeg	2017-07-07 01:34:36.930659301 +0300
@@ -24,6 +24,9 @@
 FFMPEG_TARGET := x86-linux-gcc
 endif
 
+FFMPEG_HOST   := $(ARCH)-libreelec-linux-gnu
+FFMPEG_TARGET := $(ARCH)-libreelec-linux-gnu
+
 unexport CFLAGS
 unexport LDFLAGS
 
@@ -167,7 +170,7 @@
 
 $(LIB_ROOT)/$(YASM)/.tvh_build: \
 		$(LIB_ROOT)/$(YASM)/.tvh_download
-	cd $(LIB_ROOT)/$(YASM) && $(CONFIGURE) \
+	cd $(LIB_ROOT)/$(YASM) && $(CONFIGURE) CC=$(HOST_CC) \
 		--libdir=/$(EPREFIX0)/lib
 	DESTDIR=$(EBUILDIR) \
 		$(MAKE) -C $(LIB_ROOT)/$(YASM) install
@@ -260,6 +263,7 @@
 	cd $(LIB_ROOT)/$(LIBX265)/build/linux && cmake -G "Unix Makefiles" \
 		-DCMAKE_INSTALL_PREFIX="/ffmpeg" \
 		-DENABLE_SHARED:BOOL=OFF \
+		-DEXTRA_LINK_FLAGS="-ldl" \
 		../../source
 	DESTDIR=$(EBUILDIR) \
 		$(MAKE) -C $(LIB_ROOT)/$(LIBX265)/build/linux install
@@ -293,7 +297,7 @@
 ifeq (yes,$(CONFIG_LIBVPX_STATIC))
 
 ifneq (,$(FFMPEG_TARGET))
-LIBVPX_TARGET := --target=$(FFMPEG_TARGET)
+LIBVPX_TARGET := --target=generic-gnu
 endif
 
 $(LIB_ROOT)/$(LIBVPX)/.tvh_download:
@@ -305,7 +309,7 @@
 		$(LIB_ROOT)/$(YASM)/.tvh_build \
 		$(LIB_ROOT)/$(LIBVPX)/.tvh_download
 	cd $(LIB_ROOT)/$(LIBVPX) && \
-		ASFLAGS="-DENABLE_PIC=1 -DPIC=1" $(CONFIGURE) \
+		ASFLAGS="-DENABLE_PIC=1 -DPIC=1" CROSS=$(FFMPEG_TARGET)- $(CONFIGURE) \
 		--extra-cflags="$(CFLAGS_PI)" \
 		--disable-examples \
 		--disable-docs \
@@ -384,7 +388,8 @@
 	cd $(LIB_ROOT)/$(LIBTHEORA) && \
 		CFLAGS="$(CFLAGS_PI)" $(CONFIGURE) \
 		--libdir=/$(EPREFIX0)/lib \
-		--with-ogg=$(EPREFIX) \
+		OGG_CFLAGS=-I$(EPREFIX)/include \
+		OGG_LIBS=-L$(EPREFIX)/lib \
 		--disable-examples \
 		--disable-spec \
 		$(LIBTHEORA_HOST)
@@ -432,7 +437,8 @@
 	cd $(LIB_ROOT)/$(LIBVORBIS) && \
 		CFLAGS="$(CFLAGS_PI)" $(CONFIGURE) \
 		--libdir=/$(EPREFIX0)/lib \
-		--with-ogg=$(EPREFIX)
+		OGG_CFLAGS=-I$(EPREFIX)/include \
+		OGG_LIBS=-L$(EPREFIX)/lib
 	DESTDIR=$(EBUILDIR) \
 		$(MAKE) -C $(LIB_ROOT)/$(LIBVORBIS) install
 	@touch $@
@@ -596,6 +602,8 @@
 		$(LIB_ROOT)/$(LIBMFX)/.tvh_build \
 		$(LIB_ROOT)/$(FFMPEG)/.tvh_download
 	cd $(LIB_ROOT)/$(FFMPEG) && $(CONFIGURE) \
+		--cc=$(FFMPEG_HOST)-gcc \
+		--cxx=$(FFMPEG_HOST)-g++ \
 		--disable-all \
 		--enable-gpl \
 		--extra-cflags="$(ECFLAGS)" \
