diff -Naur a/xbmc/cores/VideoPlayer/DVDCodecs/Video/AMLCodec.cpp b/xbmc/cores/VideoPlayer/DVDCodecs/Video/AMLCodec.cpp
--- a/xbmc/cores/VideoPlayer/DVDCodecs/Video/AMLCodec.cpp	2016-09-17 16:57:58.000000000 +0300
+++ b/xbmc/cores/VideoPlayer/DVDCodecs/Video/AMLCodec.cpp	2016-09-18 00:40:21.946502458 +0300
@@ -1599,7 +1599,7 @@
         am_private->gcodec.param = (void*)(EXTERNAL_PTS | SYNC_OUTSIDE);
       break;
   }
-  am_private->gcodec.param = (void *)((std::uintptr_t)am_private->gcodec.param | (am_private->video_rotation_degree << 16));
+  am_private->gcodec.param = (void *)((unsigned long)am_private->gcodec.param | (am_private->video_rotation_degree << 16));
 
   // translate from generic to firemware version dependent
   m_dll->codec_init_para(&am_private->gcodec, &am_private->vcodec);
