diff -Naur a/xbmc/addons/Addon.cpp b/xbmc/addons/Addon.cpp
--- a/xbmc/addons/Addon.cpp	2016-09-17 16:57:58.000000000 +0300
+++ b/xbmc/addons/Addon.cpp	2016-09-18 00:31:27.746514673 +0300
@@ -340,6 +340,13 @@
   return AddonVersion("0.0.0");
 }
 
+void CallOEWrapper(const std::string& ID, bool disable)
+{
+  char cmd[255];
+  snprintf(cmd, sizeof(cmd), "/usr/lib/libreelec/systemd-addon-wrapper %s %d %d", ID.c_str(), disable);
+  system(cmd);
+}
+
 void OnEnabled(const std::string& id)
 {
   // If the addon is a special, call enabled handler
@@ -348,6 +355,11 @@
       CAddonMgr::GetInstance().GetAddon(id, addon, ADDON_ADSPDLL))
     return addon->OnEnabled();
 
+  // OE: systemctl enable & start on addon enable
+  if (CAddonMgr::GetInstance().GetAddon(id, addon, ADDON_SERVICE))
+    CallOEWrapper(addon->ID(), false);
+  // OE
+
   if (CAddonMgr::GetInstance().ServicesHasStarted())
   {
     if (CAddonMgr::GetInstance().GetAddon(id, addon, ADDON_SERVICE))
@@ -370,6 +382,11 @@
   {
     if (CAddonMgr::GetInstance().GetAddon(id, addon, ADDON_SERVICE, false))
       std::static_pointer_cast<CService>(addon)->Stop();
+
+    // OE: systemctl stop & disable on addon disable
+    if (CAddonMgr::GetInstance().GetAddon(id, addon, ADDON_SERVICE, false))
+      CallOEWrapper(addon->ID(), true);
+    // OE
   }
 
   if (CAddonMgr::GetInstance().GetAddon(id, addon, ADDON_CONTEXT_ITEM, false))
@@ -399,6 +416,14 @@
 void OnPostInstall(const AddonPtr& addon, bool update, bool modal)
 {
   AddonPtr localAddon;
+  // OE: systemctl stop & disable / enable & start on addon upgrade
+  if (CAddonMgr::GetInstance().GetAddon(addon->ID(), localAddon, ADDON_SERVICE))
+  {
+    CallOEWrapper(addon->ID(), true);
+    CallOEWrapper(addon->ID(), false);
+  }
+  // OE
+
   if (CAddonMgr::GetInstance().ServicesHasStarted())
   {
     if (CAddonMgr::GetInstance().GetAddon(addon->ID(), localAddon, ADDON_SERVICE))
@@ -414,6 +439,10 @@
 void OnPreUnInstall(const AddonPtr& addon)
 {
   AddonPtr localAddon;
+  // OE: systemctl stop & disable on addon ininstall
+  if (CAddonMgr::GetInstance().GetAddon(addon->ID(), localAddon, ADDON_SERVICE))
+    CallOEWrapper(addon->ID(), true);
+  // OE
 
   if (CAddonMgr::GetInstance().ServicesHasStarted())
   {
diff -Naur a/xbmc/addons/Addon.h b/xbmc/addons/Addon.h
--- a/xbmc/addons/Addon.h	2016-09-17 16:57:58.000000000 +0300
+++ b/xbmc/addons/Addon.h	2016-09-18 00:34:24.518510631 +0300
@@ -120,6 +120,7 @@
   uint64_t PackageSize() const override { return m_props.packageSize; }
   const InfoMap& ExtraInfo() const override { return m_props.extrainfo; }
   const ADDONDEPS& GetDeps() const override { return m_props.dependencies; }
+  void CallOEWrapper(const std::string& ID, bool disable);
 
   /*! \brief Check whether the this addon can be configured or not
    \return true if the addon has settings, false otherwise
