diff -Naur a/Makefile b/Makefile
--- a/Makefile	2011-03-25 17:50:08.000000000 +0200
+++ b/Makefile	2018-05-16 16:52:02.003737260 +0300
@@ -92,7 +92,7 @@
 ### Internationalization (I18N):
 
 PODIR     = po
-LOCALEDIR = $(VDRDIR)/locale
+LOCALEDIR = $(LOCDIR)
 I18Npo    = $(wildcard $(PODIR)/*.po)
 I18Nmsgs  = $(addprefix $(LOCALEDIR)/, $(addsuffix /LC_MESSAGES/vdr-$(PLUGIN).mo, $(notdir $(foreach file, $(I18Npo), $(basename $(file))))))
 I18Npot   = $(PODIR)/$(PLUGIN).pot
