diff -Naur a/src/defaults.py b/src/defaults.py
--- a/src/defaults.py	2019-06-11 18:53:22.000000000 +0300
+++ b/src/defaults.py	2019-08-08 04:29:14.124015691 +0300
@@ -34,6 +34,55 @@
     }
 
 ################################################################################
+# Transmission Module
+################################################################################
+
+transmission = {
+    'ENABLED': True,
+    'TRANSMISSION_DAEMON': '/usr/bin/transmission-daemon',
+    'TRANSMISSION_DAEMON_START': '/usr/bin/transmission.start',
+    'D_TRANSMISSION_SECURE': '1',
+    'D_TRANSMISSION_USERNAME': 'libreelec',
+    'D_TRANSMISSION_PASSWORD': 'libreelec',
+    'D_TRANSMISSION_IP': '*.*.*.*',
+    'D_TRANSMISSION_LIMIT_GLOBAL': '100',
+    'D_TRANSMISSION_LIMIT_TORRENT': '100',
+    'D_TRANSMISSION_DL_DIR': '/storage/downloads/',
+    }
+
+################################################################################
+# Tv Servers Module
+################################################################################
+
+tvservers = {
+    'ENABLED': True,
+    'TVHEADEND_DAEMON': '/usr/bin/tvheadend',
+    'TVHEADEND_DAEMON_START': '/usr/bin/tvheadend.start',
+    'OSCAM_DAEMON': '/usr/bin/oscam',
+    'OSCAM_DAEMON_START': '/usr/bin/oscam.start',
+    #TVBACKEND
+    'D_TV_BACKEND': 'none',
+    #TVHEADEND
+    'D_PRELOAD_CAPMT_CA': '0',
+    'D_TVH_BACKUP': '0',
+    'D_TVH_DEBUG': '0',
+    'D_TVH_ADAPTERS': '',
+    'D_TVH_OTHER_ARG': '',
+    'D_XMLTV_TYPE': 'none',
+    'D_XMLTV_LOCATION_FILE': '',
+    'D_XMLTV_LOCATION_WEB': 'http://',
+    'D_XMLTV_LOCATION_SCRIPT': '',
+    #DVB
+    'D_WAIT_FOR_FEINIT': '0',
+    'D_NUM_ADAPTERS': '1',
+    'D_REMOVE_MODULES': '',
+    #OSCAM
+    'D_OSCAM_ENABLE': '1',
+    'D_OSCAM_DEBUG': '1',
+    'D_RESTART_ON_RESUME': '0',
+     }
+
+################################################################################
 # Service Module
 ################################################################################
 
@@ -54,6 +191,8 @@
     'D_SSH_DISABLE_PW_AUTH': '0',
     'AVAHI_DAEMON': '/usr/sbin/avahi-daemon',
     'CRON_DAEMON': '/sbin/crond',
+    'LCD_DRIVER_DIR': '/usr/lib/lcdproc/',
+    'D_LCD_DRIVER': 'none',
     }
 
 system = {
@@ -94,4 +233,8 @@
     'obexd': ['obex.service'],
     'crond': ['cron.service'],
     'iptables': ['iptables.service'],
+    'lcdd': ['lcdd.service'],
+    'transmission': ['transmission.service'],
+    'tvheadend': ['tvheadend.service'],
+    'oscam': ['oscam.service'],
     }
