diff -Naur a/src/resources/lib/modules/updates.py b/src/resources/lib/modules/updates.py
--- a/src/resources/lib/modules/updates.py	2019-06-11 18:53:22.000000000 +0300
+++ b/src/resources/lib/modules/updates.py	2019-08-08 03:51:36.476020295 +0300
@@ -46,25 +46,57 @@
                             'value': 'auto',
                             'action': 'set_auto_update',
                             'type': 'multivalue',
-                            'values': ['auto', 'manual'],
+                            'values': ['auto', 'manual', 'custom'],
                             'InfoText': 714,
                             'order': 1,
                             },
+                        'CustomUpdateSettings': {
+                            'name': 43000,
+                            'value': '0',
+                            'action': 'do_update_settings',
+                            'type': 'button',
+                            'parent': {
+                                'entry': 'AutoUpdate',
+                                'value': ['custom'],
+                                },
+                            'InfoText': 1700,
+                            'order': 2,
+                            },
+                        'CustomUpdateDialog': {
+                            'name': 43001,
+                            'value': '0',
+                            'action': 'do_update_dialog',
+                            'type': 'button',
+                            'parent': {
+                                'entry': 'AutoUpdate',
+                                'value': ['custom'],
+                                },
+                            'InfoText': 1701,
+                            'order': 3,
+                            },
                         'SubmitStats': {
                             'name': 32021,
                             'value': '1',
                             'action': 'set_value',
                             'type': 'bool',
+                            'parent': {
+                                'entry': 'AutoUpdate',
+                                'value': ['auto', 'manual'],
+                                },
                             'InfoText': 772,
-                            'order': 2,
+                            'order': 4,
                             },
                         'UpdateNotify': {
                             'name': 32365,
                             'value': '1',
                             'action': 'set_value',
                             'type': 'bool',
+                            'parent': {
+                                'entry': 'AutoUpdate',
+                                'value': ['auto', 'manual'],
+                                },
                             'InfoText': 715,
-                            'order': 3,
+                            'order': 5,
                             },
                         'ShowCustomChannels': {
                             'name': 32016,
@@ -76,7 +108,7 @@
                                 'value': ['manual'],
                                 },
                             'InfoText': 761,
-                            'order': 4,
+                            'order': 6,
                             },
                         'CustomChannel1': {
                             'name': 32017,
@@ -88,7 +120,7 @@
                                 'value': ['1'],
                                 },
                             'InfoText': 762,
-                            'order': 5,
+                            'order': 7,
                             },
                         'CustomChannel2': {
                             'name': 32018,
@@ -100,7 +132,7 @@
                                 'value': ['1'],
                                 },
                             'InfoText': 762,
-                            'order': 6,
+                            'order': 8,
                             },
                         'CustomChannel3': {
                             'name': 32019,
@@ -112,7 +144,7 @@
                                 'value': ['1'],
                                 },
                             'InfoText': 762,
-                            'order': 7,
+                            'order': 9,
                             },
                         'Channel': {
                             'name': 32015,
@@ -125,7 +157,7 @@
                                 },
                             'values': [],
                             'InfoText': 760,
-                            'order': 8,
+                            'order': 10,
                             },
                         'Build': {
                             'name': 32020,
@@ -137,7 +169,7 @@
                                 'value': ['manual'],
                                 },
                             'InfoText': 770,
-                            'order': 9,
+                            'order': 11,
                             },
                         }
                     }
@@ -183,6 +215,24 @@
         self.oe.dbg_log('updates::exit', 'exit_function', 0)
         pass
 
+    def do_update_settings(self, listItem=None):
+        try:
+            self.oe.dbg_log('updates::do_update_settings', 'enter_function', 0)
+            if xbmc.getCondVisibility('System.HasAddon(script.libreelec.devupdate)'):
+                xbmc.executebuiltin('Addon.OpenSettings(script.libreelec.devupdate)')
+            self.oe.dbg_log('updates::do_update_settings', 'exit_function', 0)
+        except Exception, e:
+            self.oe.dbg_log('updates::do_update_settings', 'ERROR: (' + repr(e) + ')')
+
+    def do_update_dialog(self, listItem=None):
+        try:
+            self.oe.dbg_log('updates::do_update_dialog', 'enter_function', 0)
+            if xbmc.getCondVisibility('System.HasAddon(script.libreelec.devupdate)'):
+                xbmc.executebuiltin('RunAddon(script.libreelec.devupdate)')
+            self.oe.dbg_log('updates::do_update_dialog', 'exit_function', 0)
+        except Exception, e:
+            self.oe.dbg_log('updates::do_update_dialog', 'ERROR: (' + repr(e) + ')')
+
     # Identify connected GPU card (card0, card1 etc.)
     def get_gpu_card(self):
         for root, dirs, files in os.walk("/sys/class/drm", followlinks=False):
