diff -Naur a/resources/lib/oeWindows.py b/resources/lib/oeWindows.py
--- a/resources/lib/oeWindows.py	2021-03-10 05:37:21.000000000 +0200
+++ b/resources/lib/oeWindows.py	2021-04-04 01:27:22.000000000 +0300
@@ -54,6 +54,7 @@
             }
 
         self.isChild = False
+        is_wizard = False
         self.lastGuiList = -1
         self.lastListType = -1
         if 'isChild' in kwargs:
@@ -460,6 +461,7 @@
 
         self.actions = {}
         self.wizards = []
+        is_wizard = True
         self.last_wizard = None
 
     def onInit(self):
