diff -Naur a/resources/lib/oe.py b/resources/lib/oe.py
--- a/resources/lib/oe.py	2021-03-10 05:37:21.000000000 +0200
+++ b/resources/lib/oe.py	2021-04-04 01:31:50.000000000 +0300
@@ -41,6 +41,7 @@
 conf_lock = False
 xbmcIsPlaying = 0
 input_request = False
+is_wizard = False
 dictModules = {}
 listObject = {
     'list': 1100,
@@ -265,8 +266,7 @@
         return self.cancelled
 
 def _(code):
-    wizardComp = read_setting('libreelec', 'wizard_completed')
-    if wizardComp == "True":
+    if not is_wizard:
         codeNew = __addon__.getLocalizedString(code)
     else:
         curLang = read_setting("system", "language")
