diff -Naur a/resources/lib/modules/updates.py b/resources/lib/modules/updates.py
--- a/resources/lib/modules/updates.py	2021-08-09 12:13:50.000000000 +0300
+++ b/resources/lib/modules/updates.py	2021-11-01 23:53:28.000000000 +0200
@@ -44,25 +44,57 @@
                     'value': 'auto',
                     'action': 'set_auto_update',
                     'type': 'multivalue',
-                    'values': ['auto', 'manual'],
+                    'values': ['auto', 'manual', 'custom'],
                     'InfoText': 714,
                     'order': 1,
                 },
+                'CustomUpdateSettings': {
+                    'name': 43000,
+                    'value': '0',
+                    'action': 'do_update_settings',
+                    'type': 'button',
+                    'parent': {
+                        'entry': 'AutoUpdate',
+                        'value': ['custom'],
+                        },
+                    'InfoText': 1700,
+                    'order': 2,
+                },
+                'CustomUpdateDialog': {
+                    'name': 43001,
+                    'value': '0',
+                    'action': 'do_update_dialog',
+                    'type': 'button',
+                    'parent': {
+                        'entry': 'AutoUpdate',
+                        'value': ['custom'],
+                        },
+                    'InfoText': 1701,
+                    'order': 3,
+                },
                 'SubmitStats': {
                     'name': 32021,
                     'value': '1',
                     'action': 'set_value',
                     'type': 'bool',
+                    'parent': {
+                        'entry': 'AutoUpdate',
+                        'value': ['auto', 'manual'],
+                        },
                     'InfoText': 772,
-                    'order': 2,
+                    'order': 4,
                 },
                 'UpdateNotify': {
                     'name': 32365,
                     'value': '1',
                     'action': 'set_value',
                     'type': 'bool',
+                    'parent': {
+                            'entry': 'AutoUpdate',
+                            'value': ['auto', 'manual'],
+                            },
                     'InfoText': 715,
-                    'order': 3,
+                    'order': 5,
                 },
                 'ShowCustomChannels': {
                     'name': 32016,
@@ -74,7 +106,7 @@
                         'value': ['manual'],
                     },
                     'InfoText': 761,
-                    'order': 4,
+                    'order': 6,
                 },
                 'CustomChannel1': {
                     'name': 32017,
@@ -86,7 +118,7 @@
                         'value': ['1'],
                     },
                     'InfoText': 762,
-                    'order': 5,
+                    'order': 7,
                 },
                 'CustomChannel2': {
                     'name': 32018,
@@ -98,7 +130,7 @@
                         'value': ['1'],
                     },
                     'InfoText': 762,
-                    'order': 6,
+                    'order': 8,
                 },
                 'CustomChannel3': {
                     'name': 32019,
@@ -110,7 +142,7 @@
                         'value': ['1'],
                     },
                     'InfoText': 762,
-                    'order': 7,
+                    'order': 9,
                 },
                 'Channel': {
                     'name': 32015,
@@ -123,7 +155,7 @@
                     },
                     'values': [],
                     'InfoText': 760,
-                    'order': 8,
+                    'order': 10,
                 },
                 'Build': {
                     'name': 32020,
@@ -135,7 +167,7 @@
                         'value': ['manual'],
                     },
                     'InfoText': 770,
-                    'order': 9,
+                    'order': 11,
                 },
             },
         },
@@ -191,6 +223,16 @@
     def exit(self):
         pass
 
+    @log.log_function()
+    def do_update_settings(self, listItem=None):
+        if xbmc.getCondVisibility('System.HasAddon(script.libreelec.devupdate)'):
+            xbmc.executebuiltin('Addon.OpenSettings(script.libreelec.devupdate)')
+
+    @log.log_function()
+    def do_update_dialog(self, listItem=None):
+        if xbmc.getCondVisibility('System.HasAddon(script.libreelec.devupdate)'):
+            xbmc.executebuiltin('RunAddon(script.libreelec.devupdate)')
+
     # Identify connected GPU card (card0, card1 etc.)
     @log.log_function()
     def get_gpu_card(self):
