diff -Naur a/resources/lib/defaults.py b/resources/lib/defaults.py
--- a/resources/lib/defaults.py	2021-10-14 06:24:06.000000000 +0300
+++ b/resources/lib/defaults.py	2021-11-10 21:36:10.595348301 +0200
@@ -57,6 +57,36 @@
     'D_SSH_DISABLE_PW_AUTH': '0',
     'AVAHI_DAEMON': '/usr/sbin/avahi-daemon',
     'CRON_DAEMON': '/sbin/crond',
+    'LCD_DRIVER_DIR': '/usr/lib/lcdproc/',
+    'D_LCD_DRIVER': 'none',
+    'TRANSMISSION_DAEMON': '/usr/bin/transmission-daemon',
+    'TRANSMISSION_DAEMON_START': '/usr/bin/transmission.start',
+    'D_TRANSMISSION_SECURE': '1',
+    'D_TRANSMISSION_USERNAME': 'libreelec',
+    'D_TRANSMISSION_PASSWORD': 'libreelec',
+    'D_TRANSMISSION_IP': '*.*.*.*',
+    'D_TRANSMISSION_LIMIT_GLOBAL': '100',
+    'D_TRANSMISSION_LIMIT_TORRENT': '100',
+    'D_TRANSMISSION_DL_DIR': '/storage/downloads/',
+    'TVHEADEND_DAEMON': '/usr/bin/tvheadend',
+    'TVHEADEND_DAEMON_START': '/usr/bin/tvheadend.start',
+    'D_PRELOAD_CAPMT_CA': '0',
+    'D_TVH_BACKUP': '0',
+    'D_TVH_DEBUG': '0',
+    'D_TVH_ADAPTERS': '',
+    'D_TVH_OTHER_ARG': '',
+    'D_XMLTV_TYPE': 'none',
+    'D_XMLTV_LOCATION_FILE': '',
+    'D_XMLTV_LOCATION_WEB': 'http://',
+    'D_XMLTV_LOCATION_SCRIPT': '',
+    'OSCAM_DAEMON': '/usr/bin/oscam',
+    'OSCAM_DAEMON_START': '/usr/bin/oscam.start',
+    'D_OSCAM_ENABLE': '1',
+    'D_OSCAM_DEBUG': '1',
+    'D_RESTART_ON_RESUME': '0',
+    'D_WAIT_FOR_FEINIT': '0',
+    'D_NUM_ADAPTERS': '1',
+    'D_REMOVE_MODULES': '',
     }
 
 system = {
@@ -93,6 +123,8 @@
     'RPI_FLASHING_TRIGGER': '/storage/.rpi_flash_firmware',
     }
 
+drivers = {'ENABLED': True}
+
 about = {'ENABLED': True}
 
 _services = {
@@ -103,4 +135,8 @@
     'obexd': ['obex.service'],
     'crond': ['cron.service'],
     'iptables': ['iptables.service'],
+    'lcdd': ['lcdd.service'],
+    'transmission': ['transmission.service'],
+    'tvheadend': ['tvheadend.service'],
+    'oscam': ['oscam.service'],
     }
