Index: b/definitions.h
===================================================================
--- a/definitions.h
+++ b/definitions.h
@@ -234,7 +234,7 @@
 };
 
 enum class eDetailedHeaderST {
-    title = eScraperHeaderST::count,
+    title = (int)eScraperHeaderST::count,
     shorttext,
     start,
     stop,
@@ -249,7 +249,7 @@
 };
 
 enum class eDetailedHeaderIT {
-    daynumeric = eScraperHeaderIT::count,
+    daynumeric = (int)eScraperHeaderIT::count,
     month,
     year,
     running,
@@ -375,7 +375,7 @@
 };
 
 enum class eDetailedEpgST {
-    title = eScraperST::count,
+    title = (int)eScraperST::count,
     shorttext,
     description,
     start,
@@ -393,7 +393,7 @@
 };
 
 enum class eDetailedEpgIT {
-    daynumeric = eScraperIT::count,
+    daynumeric = (int)eScraperIT::count,
     month,
     year,
     running,
@@ -543,4 +543,4 @@
     count
 };
 
-#endif //__DEFINITIONS_H
\ No newline at end of file
+#endif //__DEFINITIONS_H
