diff -Naur a/modules/codec/aom.c b/modules/codec/aom.c
--- a/modules/codec/aom.c	2018-12-17 16:30:21.068085858 +0200
+++ b/modules/codec/aom.c	2018-12-17 16:32:47.201435650 +0200
@@ -96,7 +96,6 @@
     { VLC_CODEC_I444, AOM_IMG_FMT_I444, 8, 0 },
 
     { VLC_CODEC_YV12, AOM_IMG_FMT_YV12, 8, 0 },
-    { VLC_CODEC_YUVA, AOM_IMG_FMT_444A, 8, 0 },
 
     { VLC_CODEC_GBR_PLANAR, AOM_IMG_FMT_I444, 8, 1 },
     { VLC_CODEC_GBR_PLANAR_10L, AOM_IMG_FMT_I44416, 10, 1 },
